File: lab-1/domain/podcast.py
from domain.media_item import media_item


class Podcast(media_item):
    def __init__(self, title, creation_date, duration, podcaster):
        super().__init__(title, creation_date)
        self.duration = duration
        self.podcaster = podcaster

    def play(self):
        print(f"Playing the podcast: {self.title} with {self.podcaster}")

    def display_info(self):
        print(f"Podcast: {self.title} with {self.podcaster}")
        print(f"Duration: {self.duration} minutes")
        print(f"Release date: {self.creation_date}")

File: src/main/kotlin/media_item/implementations/Podcast.kt
package media_item.implementations
import media_item.MediaItem
import media_item_implementation.MediaItemImplementation
import media_item_implementation.implementations.PodcastImplementation

class Podcast(
        override val title: String,
        override val creationDate: String,
        val duration: Float,
        val podcaster: String,
        override val implementation: PodcastImplementation
) : MediaItem() {

    override fun displayInfo() {
        println("Podcast: $title with $podcaster")
        println("Duration: $duration minutes")
        println("Release date: $creationDate")
    }

    override fun toString(): String {
        return "Podcast(title='$title', creationDate='$creationDate', duration=$duration, podcaster='$podcaster')"
    }
}